services/reminders/reminders.py

```diff
@@ -0,0 +1,141 @@
+
+from baselayer.log import make_log
+from baselayer.app.models import init_db
+from baselayer.app.env import load_env
+from baselayer.app.flow import Flow
+from skyportal.models import (
+    DBSession,
+    Reminder,
+    ReminderOnSpectrum,
+    ReminderOnGCN,
+    ReminderOnShift,
+    UserNotification,
+)
+from skyportal.models.gcn import GcnEvent
+from skyportal.models.shift import Shift
+import time
+
+env, cfg = load_env()
+
+init_db(**cfg['database'])
+
+log = make_log('reminders')
+
+REQUEST_TIMEOUT_SECONDS = cfg['health_monitor.request_timeout_seconds']
+
+
+def service():
+    loaded = False
+    while not loaded:
+        # ping the app to see if its running
+        port = cfg['ports.app_internal']
+        try:
+            r = requests.get(
+                f'http://localhost:{port}/api/sysinfo', timeout=REQUEST_TIMEOUT_SECONDS
+            )
+        except:  # noqa: E722
+            status_code = 0
+        else:
+            status_code = r.status_code
+
+        if status_code == 200:
+            loaded = True
+    while loaded:
+        try:
+            send_reminders()
+        except Exception as e:
+            log(e)
+
+
+def send_reminders():
+    now = datetime.utcnow()
+    reminders = []
+    with DBSession() as session:
+        try:
+            reminders = (
+                (
+                    session.query(Reminder)
+                    .where(Reminder.number_of_reminders > 0)
+                    .where(Reminder.next_reminder <= now)
+                    .all()
+                )
+                + (
+                    session.query(ReminderOnSpectrum)
+                    .where(ReminderOnSpectrum.number_of_reminders > 0)
+                    .where(ReminderOnSpectrum.next_reminder <= now)
+                    .all()
+                )
+                + (
+                    session.query(ReminderOnGCN)
+                    .where(ReminderOnGCN.number_of_reminders > 0)
+                    .where(ReminderOnGCN.next_reminder <= now)
+                    .all()
+                )
+                + (
+                    session.query(ReminderOnShift)
+                    .where(ReminderOnShift.number_of_reminders > 0)
+                    .where(ReminderOnShift.next_reminder <= now)
+                    .all()
+                )
+            )
+        except Exception as e:
+            log(e)
+
+        ws_flow = Flow()
+        for reminder in reminders:
+            reminder_type = reminder.__class__
+            if reminder_type == Reminder:
+                text_to_send = (
+                    f"Reminder of source *{reminder.obj_id}*: {reminder.text}"
+                )
+                url_endpoint = f"/source/{reminder.obj_id}"
+            elif reminder_type == ReminderOnSpectrum:
+                text_to_send = (
+                    f"Reminder of spectrum *{reminder.spectrum_id}*: {reminder.text}"
+                )
+                url_endpoint = f"/source/{reminder.spectrum_id}"
```

Comment:
I don't think there is any routing to a spectrum page right? I think that we want the user to be redirected to the source's page that will have that spectrum on it.  But then it might be incorrect to have the spectrum_id, but we should have an obj_id at the end of the URL. I'll look into that.

Reply:
There's a way to (maybe) direct to a spectrum via the "manage data" page. 
Although there could be a way to navigate directly to the spectroscopy plot (I'm not sure how you'd do that).

Reply:
No idea how to directly navigate to the spectroscopy plot. I'll look into that. If I can't figure it out, I'll go with the first option.

Reply:
Adding some anchors means adding additional routing, which can easily get complicated.

Reply:
Same with the manage_data, we need some routing if we want to directly open the right spectra and such. Should this be done in this PR?

Reply:
Not critical. If it isn't very easy to do, just leave it pointing at the source page. Just make sure the link uses the `obj_id` not the `spectrum_id`.

Reply:
Ok! Will do